### ISSUE TITLE
RangeSelector

### ISSUE THREAD

**Issue by glenrobson** [IIIF/presentation-validator#172]
Current this manifest is failing validation. It has the following annotation structure:

```
    "target": {
        "source": {
            "id": ["https://tanyaclement.github.io/znh-1939/field_recordings_t86_243/canvas-2/canvas"](https://tanyaclement.github.io/znh-1939/field_recordings_t86_243/canvas-2/canvas),
            "type": "Canvas",
            "partOf": [{
                "id": ["https://tanyaclement.github.io/znh-1939/manifests.json"](https://tanyaclement.github.io/znh-1939/manifests.json),
                "type": "Manifest"
             }]
        },
        "selector": {
            "type": "RangeSelector",
            "t": "52,60"
        }
    }
```

Range selector defined in W3C:

https://www.w3.org/TR/annotation-model/#range-selector

Is `t` valid here? The W3C says that it needs to have a `startSelector` and `endSelector`. Should it use a fragement selector? or a PointSelector?

**Comment by saracarl** [comment]
@azaroth42 Could you weigh in here?  What is the right way for an annotation to target a time range on a canvas containing time based media like an audio or video file?  

I think the above example was probably derived (or copied from elsewhere?), likely incorrectly, from the PointSelector format:
```
"selector": {
            "type": "RangeSelector",
            "t": "52,60"
        }
```

@glenrobson's point is that the W3C Annotation model says a [RangeSelector needs a startSelector and an endSelector](https://www.w3.org/TR/annotation-model/#range-selector):
```
"selector": {
      "type": "RangeSelector",
      "startSelector": {
        "type": "XPathSelector",
        "value": "//table[1]/tr[1]/td[2]"
      },
      "endSelector": {
        "type": "XPathSelector",
        "value": "//table[1]/tr[1]/td[4]"
      }
    }
```

Then the question is "what is the `type` and `value`; the obvious `type` is a [MediaFragment](https://www.w3.org/TR/media-frags/#naming-time), but that has 2 problems:
a) A MediaFragment (i.e. `t=52,60`) has a start and stop built into it, so the data doesn't match the `startSelector` and `endSelector` model from the Annotations model.  (It's not a PointSelector.)
b) I don't think IIIF manifests should be targeting the media file itself, which MediaFragment seems to imply.  We should be targeting a range of the canvas.  In other words, what do we tack the `t=52,60` onto?  Perhaps the canvas?  `https://tanyaclement.github.io/znh-1939/field_recordings_t86_243/canvas-2/canvas#t=52,60`

**Comment by kirschbombe** [comment]
@saracarl - I'm no Rob, but I think you would want to target the canvas and not the media file, just like you have in the last example. I thought we might have an example in the cookbook, but didn't see one at first glance.  `.../canvas#t=52,60` should validate, although may not work in current viewers (maybe Theseus?).

**Comment by azaroth42** [comment]
Yes, RangeSelector is for designating a start and end point, and then the selection is between those two points. When you have a fragment like `t=52,60` you can just use the regular FragmentSelector :)

Also, yes, given that it's a media fragment, you could simplify even further as Dawn says by just putting it on the end of the canvas URI.

**Comment by saracarl** [comment]
OK.  That makes sense to me.

It took me forever to find this in the spec, as it's in the middle of the [`canvas` section](https://iiif.io/api/presentation/3.0/#53-canvas) rather than the annotation section.  I'm copying it here for reference.

_Parts of Canvases may be described using a Specific Resource with a Selector, following the patterns defined in the [Web Annotation](http://w3.org/TR/annotation-model/) data model. The use of the FragmentSelector class is recommended by that specification, as it allows for refinement by other Selectors and for consistency with use cases that cannot be represented using a URI fragment directly. Parts of Canvases can be [referenced](https://iiif.io/api/presentation/3.0/#12-terminology) from Ranges, as the body or target of Annotations, or in the start property.

Parts of Canvases may also be identified by appending a fragment to the Canvas’s URI, and these parts are still considered to be Canvases: their type value is the string Canvas. Rectangular spatial parts of Canvases may also be described by appending an xywh= fragment to the end of the Canvas’s URI. Similarly, temporal parts of Canvases may be described by appending a t= fragment to the end of the Canvas’s URI. Spatial and temporal fragments may be combined, using an & character between them, and the temporal dimension should come first. It is an error to select a region using a dimension that is not defined by the Canvas, such as a temporal region of a Canvas that only has height and width dimensions._
